### PR TITLE
Feature/mqtt stuff

### PR DESCRIPTION
**Pull request by SgtSilvio**
Added structure for MQTT Agents.

### CODE REVIEW

File: src/main/java/org/varroa/load/AbstractMqttAgent.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ */
+public abstract class AbstractMqttAgent implements MqttAgent {
+
+    @Override
+    public @NotNull Completable execute(final @NotNull Chunk chunk) {
+        return executeCommands(chunk.getCommands());
+    }
+
+    private @NotNull Completable executeCommands(final @NotNull ImmutableList<Command> commands) {
+        Completable lifecycle = Completable.complete();
```
Comment (s-gheldd) [inline src/main/java/org/varroa/load/AbstractMqttAgent.java]: final

File: src/main/java/org/varroa/load/AbstractMqttAgent.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                    break;
+                case WAIT:
+                    lifecycle = lifecycle.andThen(wait((WaitCommand) command));
+                    break;
+            }
+        }
+        return lifecycle;
+    }
+
+    abstract @NotNull Completable connect(@NotNull ConnectCommand connectCommand);
```
Comment (s-gheldd) [inline src/main/java/org/varroa/load/AbstractMqttAgent.java]: final is missing in all method parameters

File: src/main/java/org/varroa/remote/agent/AgentHandshakeHandler.java [hunk 3]
```diff
@@ -27,15 +31,22 @@
     static final @NotNull String NAME = "agent.handshake";
 
     private final @NotNull Map<Integer, ChunkMessage> chunkMap = new HashMap<>();
-    private final @NotNull Map<String, MqttClientPool> mqttClientPoolMap = new HashMap<>();
+    private final @NotNull Map<String, MqttAgentPool> mqttClientPoolMap = new HashMap<>();
+
+    private @Nullable Scheduler nettyEventLoopScheduler;
+
+    @Override
+    public void handlerAdded(final ChannelHandlerContext ctx) {
```
Comment (s-gheldd) [inline src/main/java/org/varroa/remote/agent/AgentHandshakeHandler.java]: @NotNull

File: src/main/java/org/varroa/remote/agent/AgentHandshakeHandler.java [hunk 4]
```diff
@@ -73,57 +84,73 @@ private void reject(final @NotNull ChannelHandlerContext ctx, final int id) {
...
         } else {
-            final MqttClientPool mqttClientPool = mqttClientPoolMap.get(chunk.getClientGroupId());
-            mqttClientGroup = mqttClientPool.take(chunk.getCount());
+            final MqttAgentPool mqttAgentPool = mqttClientPoolMap.get(chunk.getClientGroupId());
+            mqttAgentGroup = mqttAgentPool.take(chunk.getCountOfClients());
         }
 
-        mqttClientGroup.execute(chunkMessage);
-    }
+        assert nettyEventLoopScheduler != null;
```
Comment (s-gheldd) [inline src/main/java/org/varroa/remote/agent/AgentHandshakeHandler.java]: please add assertion message

File: src/main/java/org/varroa/load/AbstractMqttAgent.java (hunk 1 above)
Comment (SgtSilvio) [reply src/main/java/org/varroa/load/AbstractMqttAgent.java]: can not be final

File: src/main/java/org/varroa/load/AbstractMqttAgent.java (hunk 2 above)
Comment (SgtSilvio) [reply src/main/java/org/varroa/load/AbstractMqttAgent.java]: because it is abstract

**Pull request merged**